cg/services/order_validation_service/rules/case_sample/utils.py

```diff
@@ -294,3 +296,17 @@ def is_invalid_plate_well_format(sample: Sample) -> bool:
     if sample.is_on_plate:
         return not bool(re.match(correct_well_position_pattern, sample.well_position))
     return False
+
+
+def is_tube_container_name_redundant(sample: Sample, counter: Counter) -> bool:
+    """Check if a tube container name is reused across samples."""
+    return sample.container == ContainerEnum.tube and counter.get(sample.container_name) > 1
+
+
+def get_repeated_tube_names(case: Case) -> list[int]:
```

Comment:
The naming here could be slightly better since it currently implies that a list of names would be returned, but in fact a list of indices are returned. Maybe `get_indices_for_repeated_tube_names`?